### PR TITLE
TINKERPOP-2533 Added gremlin-language module.

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-2533

This module is currently standalone, meaning it depends on no other TinkerPop modules nor does any module depend on it. It represents an initial body of work which has reasonable coverage of Gremlin and a expansive and flexible test suite.

Builds with `mvn clean install`

VOTE +1

### CODE REVIEW

File: docs/src/recipes/recommendation.asciidoc [hunk 1]
```diff
@@ -33,7 +33,7 @@ g.addV("person").property("name","alice").
   addV("person").property("name","bob").
   addV("person").property("name","jon").
   addV("person").property("name","jack").
-  addV("person").property("name","jill")iterate()
+  addV("person").property("name","jill").iterate()
```
Comment (FlorianHockmann) [inline docs/src/recipes/recommendation.asciidoc:36]: Nice to see that this already uncovered some bugs in our docs :)

**Review by danielmartincraig — COMMENTED** [review]
I love this update!

**Comment by spmallette** [comment]
@danielmartincraig glad to hear that you like this direction. do you happen to have something specific in mind that you might do with the grammar?

**Comment by danielmartincraig** [comment]
No I don't, but I'm following the discussion on this topic with interest.

**Pull request merged**